### ISSUE TITLE
NaN Tensor Values Error

### ISSUE THREAD

**Issue by singhanj13** [tryolabs/luminoth#133]
I have checked that all my images have valid bounding boxes (within the image boundaries). Yet, whenever I start training, after 30-60 steps I face the following error:

**InvalidArgumentError (see above for traceback): Nan in summary histogram for: fasterrcnn/rpn/rpn_bbox_target/histogram**
	 [[Node: fasterrcnn/rpn/rpn_bbox_target/histogram = HistogramSummary[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"](fasterrcnn/rpn/rpn_bbox_target/histogram/tag, fasterrcnn/rpn/anchor_target/ScatterNd)]]

1) Could this be fixed by changing the learning rate and/or batch size? 
2) Many of my images have some very small objects. How small objects (occupying how much of the total area) can be identified by the object detection code?

**Comment by IanTayler** [comment]
Hi @singhanj13!

Were you able to fix the issue? It would be nice to know what was the problem and the solution to see if we can improve Luminoth!

**Comment by IanTayler** [comment]
About your second question, the model using the base config should be able to detect fairly small objects. If you're getting bad results with small objects and want to improve the recall on them, you can change `model.anchors.scales`.

By default, `model.anchors.scales` is `[0.5, 1, 2]`, you can change it to something like `[0.25, 0.5, 1]` if you want your model to focus on small objects. You will have to retrain it from scratch if you change that value of the config, though.

**Comment by singhanj13** [comment]
@IanTayler  I think the issue was closed by mistake. I changed the anchor size as you had suggested, because of which I faced the error at a much later stage in my training. So I wasn't able to fix the issue. I am still getting the NaN Tensor Values Error.  I have double checked my annotations and there is nothing wrong with them. Any idea what I can do to get rid of this error?

**Comment by IanTayler** [comment]
If it improved by adding smaller scales, then it may be a problem caused by very small objects. I'll look into it.

**Comment by IanTayler** [comment]
Hi, singhanj13! We found some cases where predicted bbox coordinates got too big and overflowed. There's several ways to fix the issue. We're studying the causes of the problem before we merge any specific solution, though.

In the meantime, to help avoid it you can set `dataset.data_augmentation=False` and make sure your learning rate is not too large (should be around `1.e-3` or smaller. If in doubt, use the default values). It may also help to set `dataset.image_preprocessing.min_size=300` and `dataset.image_preprocessing.max_size=900`.
